### ISSUE TITLE
How to crop padding space?

### ISSUE THREAD

**Issue by dejurin** [svg/svgo#1215]
<img width="1551" alt="Снимок экрана 2020-02-03 в 21 28 40" src="https://user-images.githubusercontent.com/4470015/73684133-3e41e500-46cc-11ea-8666-f7e1da0ea7c9.png">

**Comment by Investigamer** [comment]
I know this is really old, but I have been scouring the world trying to find a way to do this with SVGO or some other NPM package. The only two npm related solutions I've found appear to be abandoned. Literally the only thing I've managed to use with real success is Inkscape from CLI, but its not exactly ideal. I found an old thread related to this question in SVGO's issues but it was dismissed as out of project scope. How is fitting the viewbox to the shape and eliminating extraneous margins not considered a potential optimization idea? If its been done in the past, it can be done again and it should be integrated into this project as an official plugin.

**Comment by GreLI** [comment]
It's about editing than something related to optimization and SVGO. But you can make viewBox smaller to focus 'window' of what SVG image shows. Or use some editor to crop images/decrease canvas.

**Comment by AmrAlSayed0** [comment]
It would be great if SVGO could do this. UI/UX designers usually handoff svg files with annoying padding around it and it forces me to either go through each svg file and check and remove the padding or use it as is in the project but account for the padding in EVERY SINGLE IMAGE!!

SVGO could follow all the paths and determine the min and max X and Ys and crop the svg to be that. Shouldn't be hard.

**Comment by GreLI** [comment]
@AmrAlSayed0 well, you can write a custom plugin for that. There are some caveats, however, with strokes that can have different joints types and filters. So, writing such general purpose plugin isn't that easy task, aside it's not about optimizing and there are little support for developing SVGO.

But isn't it easier to ask designers to deliver better suited svg files? Also, if you are exporting svg from Figma, there are ways to select better frames. You can see them in the left panel.

**Comment by GreLI** [comment]
Also, there are some tips for doing something like that at https://github.com/svg/svgo/issues/2005#issuecomment-2168264317

**Comment by xamgore** [comment]
> @GreLI: It's about editing than something related to optimization and SVGO.

Well, it's quite interesting that we can optimize SVG while cropping it. That happens due to the translation of all paths—its coordinates—from the corner closer to the `(0, 0)` point. With a certain degree of luck, all the coordinates lose a digit then.

<img width="500" alt="illustration" src="https://github.com/user-attachments/assets/63b36a28-4981-4919-b280-4e87147a3b43" />

This approach was implemented in SVGO plugin [xamgore/svgo-autocrop](https://github.com/xamgore/svgo-autocrop). It could be improved by rescaling technique (it is possible to apply a scale factor to each coordinate to fit them into a 0-9, 0-99 range, or trim trailing digits after a dot).

**Comment by GreLI** [comment]
@xamgore Yep, people can crop image and then optimize in with SVGO. Altering images is not SVGO task, but editors. SVGO is not an editor, it doesn't make sense for the tool to alter images. People might not expect this, especially in automated pipeline. E.g. it would not be nice to see that the image has been changed after it was published on some site. Ideally you just pass an image one time and get an optimized version with less file size but still looking same. If you need to edit, just do it in an editor.

**Comment by SethFalco** [comment]
I strongly agree with GreLI. SVGO is an optimizer, not an editor. We shouldn't commit to maintaining plugins that are out of scope.

As others have reported, one can already do this through Inkscape CLI, and we have a flexible plugin API where one can do it themselves through SVGO, or even use off the shelf packages.

Yes, cropping can be an optimization. In the same way removing shapes could be an optimization too. But those are editorial decisions first and foremost, not optimizations—it's a different problem than what SVGO aims to solve!
